### PR TITLE
UIOR-285 Refine the POL notes handling

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Refine the POL notes handling
https://issues.folio.org/browse/UIOR-285
## Approach
Usage:
- `NoteCreatePage`
- `NoteEditPage`
- `NoteViewPage`
- `NotesSmartAccordion`

from '@folio/stripes/smart-components';
## ScreenShots
image

image

image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: test/bigtest/tests/POLine/line-notes-test.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      compositePoLines: [line.attrs],
+      id: line.attrs.purchaseOrderId,
+    });
+
+    const noteType = this.server.create('note-type');
+
+    this.server.create('note', {
+      type: noteType.name,
+      typeId: noteType.id,
+      links: [{ type: 'poLine', id: line.id }],
```
Comment (aliaksei-chumakou) [inline test/bigtest/tests/POLine/line-notes-test.js]: shouldn't these types be a constants?

File: translations/ui-orders/en.json [hunk 2]
```diff
@@ -760,5 +759,8 @@
   "requests.message.itemId": "Item ({itemId})",
   "requests.footer.close": "Close",
   "requests.request": "Request",
-  "requests.request.isOpened": "Yes"
+  "requests.request.isOpened": "Yes",
+
+  "poLine": "PO line",
+  "poLine.pluralized": "{count} {count, plural, one {PO line} other {PO lines}}"
```
Comment (aliaksei-chumakou) [inline translations/ui-orders/en.json]: some brainfuck here...

Comment (AndreiShumski2) [reply translations/ui-orders/en.json]: react-intl [doc](https://github.com/formatjs/react-intl)
the same in [e-holdings](https://github.com/folio-org/ui-eholdings/blob/e6adc9b5a757a99d4749ebb541bd98aa11e0e828/translations/ui-eholdings/en.json#L354) & [users](https://github.com/folio-org/ui-users/blob/964a701304344ba900235f8a52464e3f6c5d175e/translations/ui-users/en.json#L3) :)

File: src/common/Notes/NoteCreate.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  if (!state) {
+    return <Redirect to={ORDERS_ROUTE} />;
+  }
+
+  return (
+    <NoteCreatePage
+      domain={ORDERS_DOMAIN}
+      entityTypeTranslationKeys={{ poLine: 'ui-orders.poLine' }}
+      navigateBack={goBack}
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteCreate.js]: `goBack`, really?

File: src/common/Notes/NoteCreate.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    : null;
+
+  if (!state) {
+    return <Redirect to={ORDERS_ROUTE} />;
+  }
+
+  return (
+    <NoteCreatePage
+      domain={ORDERS_DOMAIN}
+      entityTypeTranslationKeys={{ poLine: 'ui-orders.poLine' }}
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteCreate.js]: please move object to constants

Comment (NikitaSedyx) [reply src/common/Notes/NoteCreate.js]: I mean make it a constant above component

File: src/common/Notes/NoteCreate.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import {
+  ORDERS_DOMAIN,
+  ORDERS_ROUTE,
+} from '../constants';
+
+const NoteCreate = ({
+  history: { goBack },
+  location: { state },
+}) => {
+  const referredRecordData = state
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteCreate.js]: it's always new object

File: src/common/Notes/NoteEdit.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      history.replace({
+        pathname: `${NOTES_ROUTE}/${id}`,
+        state,
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [id],
+  );
+
+  const referredEntityData = state
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteEdit.js]: looks like copy paste

File: src/common/Notes/NoteEdit.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      name: state.entityName,
+      type: state.entityType,
+      id: state.entityId,
+    }
+    : null;
+
+  return (
+    <NoteEditPage
+      domain={ORDERS_DOMAIN}
+      entityTypePluralizedTranslationKeys={{ poLine: 'ui-orders.poLine.pluralized' }}
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteEdit.js]: same here

File: src/common/Notes/NoteEdit.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      paneHeaderAppIcon="orders"
+      referredEntityData={referredEntityData}
+    />
+  );
+};
+
+NoteEdit.propTypes = {
+  history: ReactRouterPropTypes.history.isRequired,
+  location: ReactRouterPropTypes.location.isRequired,
+  match: PropTypes.shape({
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteEdit.js]: why not just match prop type?

File: src/common/Notes/NoteCreate.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (NikitaSedyx) [inline src/common/Notes/NoteCreate.js]: is it really common? it's used only for POL, it looks like generic, but strictly connected to the route name

File: src/common/Notes/NoteCreate.js (hunk 3 above)
Comment (AndreiShumski2) [reply src/common/Notes/NoteCreate.js]: `if (!state) { return <Redirect to={ORDERS_ROUTE} />; }`
see above)

File: src/common/Notes/NoteCreate.js (hunk 9 above)
Comment (AndreiShumski2) [reply src/common/Notes/NoteCreate.js]: I think purchase orders will have the same notes accordion

**Comment by AndreiShumski2** [comment]
Coverage on new code is more than 78%. Only conditions from common components are remained as not covered

**Pull request merged**